src/Storages/StorageMergeTree.cpp

```diff
@@ -885,6 +886,32 @@ std::shared_ptr<StorageMergeTree::MergeMutateSelectedEntry> StorageMergeTree::se
             commands.insert(commands.end(), it->second.commands.begin(), it->second.commands.end());
         }
 
+        bool is_partition_affected = false;
+        for (const auto & command : commands)
+        {
+            if (command.partition == nullptr)
+            {
+                is_partition_affected = true;
+                break;
+            }
+
+            const String partition_id = part->storage.getPartitionIDFromQuery(command.partition, global_context);
+            if (partition_id == part->info.partition_id)
+            {
+                is_partition_affected = true;
+                break;
+            }
+        }
+
+        if (!is_partition_affected)
+        {
+            /// Shall not create a new part, but will do that later if mutation with higher version appear.
+            auto block_range = std::make_pair(part->info.min_block, part->info.max_block);
+            updated_version_by_block_range[block_range] = current_mutations_by_version.rbegin()->first;
+            commands.clear();
```

Comment:
Also this place looks a little bit unsafe, maybe it worth moving `commands` into the `for` loop scope?

Reply:
+